src/highdicom/map/sop.py

```diff
@@ -0,0 +1,700 @@
+    ) -> None:
+        """Add values to the parametric map.
+
+        Parameters
+        ----------
+        pixel_array: np.ndarray
+            Array of parametric map pixel data of unsigned integer or
+            floating-point data type representing one or more frames of the
+            parametric map pixel data. The values are supposed to represent a
+            single "feature", i.e., be the result of one set of image
+            transformations such that the same `real_world_value_mappings`
+            apply.
+        real_world_value_mappings: Union[Sequence[highdicom.map.RealWorldValueMappingSequence], None], optional
+            Description of the mapping of values stored in `pixel_array` to
+            real-world values.
+        plane_positions: Sequence[highdicom.PlanePositionSequence], optional
+            Position of each plane in `pixel_array` relative to the
+            patient or slide coordinate system.
+
+        Raises
+        ------
+        ValueError
+            [description]
+        ValueError
+            [description]
+        NotImplementedError
+            [description]
+
+        """  # noqa
+        if pixel_array.ndim == 2:
+            pixel_array = pixel_array[np.newaxis, ...]
+        if pixel_array.ndim != 3:
+            raise ValueError('Pixel array must be a 2D or 3D array.')
+
+        if pixel_array.shape[1:3] != (self.Rows, self.Columns):
+            raise ValueError(
+                'Pixel array has the wrong number of rows or columns.'
+            )
+
+        if len(real_world_value_mappings) == 0:
+            raise ValueError(
+                'At least one RealWorldValueMapping item must be provided.'
+            )
+
+        if plane_positions is None:
+            if pixel_array.shape[0] != len(self._source_plane_positions):
+                raise ValueError(
+                    'Number of pixel array planes does not match number '
+                    'of planes (frames) in referenced source image.'
+                )
+            plane_positions = self._source_plane_positions
+        else:
+            if pixel_array.shape[0] != len(plane_positions):
+                raise ValueError(
+                    'Number of pixel array planes does not match number of '
+                    'provided plane positions.'
+                )
+
+        for i in range(pixel_array.shape[0]):
+            self.NumberOfFrames += 1
+
+            # Per-frame Functional Groups
+            pffg_item = Dataset()
+            pffg_item.DerivationImageSequence = []
+
+            # Plane Position (Patient/Slide)
+            if self._coordinate_system == CoordinateSystemNames.SLIDE:
+                pffg_item.PlanePositionSlideSequence = plane_positions[i]
+            else:
+                pffg_item.PlanePositionSequence = plane_positions[i]
+
+            # Frame Content
+            frame_content_item = Dataset()
+            # FIXME
```

Comment:
What needs to be fixed?

Reply:
Thanks @cgorman for pointing that out.

@CPBridge we should discuss the implementation of Dimension Index Values. It's currently wrong in Segmentation and we'll have to fix it here, too.

Reply:
yeah, I still haven't started this. Hoping to get to it sometime in the next week or so. I'll try and make the fix applicable here too if I can

Reply:
Ok. Thanks @CPBridge. I'll remove the FIXME comment and create an issue instead to keep track of it.

Reply:
@CPBridge I only included the spatial dimensions into the Dimension Index Sequence, since I didn't find an attribute that could be indexed for a particular set of real world value mappings (image transformation). I initially considered using the `RealWorldValueMappingSequence.LUTLabel`, but that has an incompatible type. Dimension Index Values has VR UL (Unsigned Short). Furthermore, the real world value mapping may be shared by all frames and consequently end up in the Shared Functional Groups Sequence rather the Per Frame Functional Groups Sequence.

@dclunie are you aware of a required attribute that could be included in the index?